Authentication/SessionServiceType.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  SessionServiceType.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/7/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+import enum Result.NoError
+
+
+public enum SessionServiceEvent<User: UserType> {
+    case LogIn(User)
+    case LogOut(User)
+    case SignUp(User)
+    case LogInError(SessionServiceError)
+    case RegistrationError(SessionServiceError)
+}
+
+public enum SessionServiceError: ErrorType {
+    case InvalidLogInCredentials(NSError?)
+    case InvalidSignUpCredentials(NSError?)
+    case NetworkError(NSError)
+}
+
+public protocol SessionServiceType {
+    
+    associatedtype User: UserType
+    
+    var currentUser: AnyProperty<User?> { get }
+    
+    var events: Signal<SessionServiceEvent<User>, NoError> { get }
+    
+    /**
+        This method takes care of validating and logging in.
+     
+        - Returns: A SignalProducer that can send the User logged in
+        or the SessionServiceError if not.
+        If the credentials are valid, the SignalProducer returned
+        must take care of:
+            sending the user to the observers
+            sending the login event through the events signal and
+            updating the currentUser property.
+        If the credentials aren't valid, the SignalProducer returned
+        must take care of:
+            sending the error to the observer and
+            sending the login error event through the events signal.
+    */
+    func logIn(email: Email, _ password: String) -> SignalProducer<User, SessionServiceError>
+    
+    /**
+         This method takes care of validating and signing up.
+         
+         - Returns: A SignalProducer that can send the User logged in
+         or the SessionServiceError if not.
+         If the credentials are valid, the SignalProducer returned
+         must take care of:
+             sending the user to the observers
+             sending the sign up and the login event through the events signal and
+             updating the currentUser property.
+         If the credentials aren't valid, the SignalProducer returned
+         must take care of:
+             sending the error to the observer and
+             sending the singup error event through the events signal.
+    */
+    func signUp(name: String, _ email: Email, _ password: String) -> SignalProducer<User, SessionServiceError>
```

Comment:
The problem was we how the first label works in Swift 2.2 which has been fixed in Swift 3.0 We can go back to full labels.